### ISSUE TITLE
Idea: Shift-Mouseclick on resources in site-tree opens in a popup like "manage elements"

### ISSUE THREAD

**Issue by Deesen** [modxcms/evolution#861]
Good idea, yes or no? Always 80% of the resources locked ;-) ..

![shift-tree](https://cloud.githubusercontent.com/assets/8569221/19699233/cf924736-9af2-11e6-9e92-3a97dfbf0b5e.png)

**Comment by dmi3yy** [comment]
no.    
better do like this:  
https://github.com/extras-evolution/EvoModal/
www.youtube.com/watch?v=8SxcnILzQa0



--  
Lukianenko Dmytro
Web developer  
Site:       http://dmi3yy.com
Skype:   dmi3yy
Phone:  +38 (091) 357-10-80


вторник, 25 октября 2016 г. в 21:37, Deesen написал:

**Comment by Deesen** [comment]
Overkill :-).. I was just thinking maybe a separate popup / browser-window is also nice to have, so you can move it on desktop.

**Comment by pmfx** [comment]
I like your idea @Deesen Nice, clean and fast way to edit/compare two resources at the same time. 
@dmi3yy I think modals from your example are overkill too.

**Comment by Nicola1971** [comment]
+ 1 really needed for me 

- modal: more cool and modern
- pop up: more usefull  - you can split windows or move to second monitor

**Comment by Deesen** [comment]
I also noticed, generally only 1 resource is locked per user. When editing in 2 windows, only latest opened resource is marked as locked, which could lead to unnessecary problems. But happens also when using browsers-tabs for editing resources.

**Comment by pmfx** [comment]
Sounds like a bug to me.

**Comment by Deesen** [comment]
I guess it is to avoid having too many resources locked. The only solution would be to remove locks when you pressed "Save and Leave" or "Cancel". If people click "Save and Stay" and then open a new resource, the previous resource would still be locked.

**Comment by bossloper** [comment]
I dont think the 1 resource lock per user is a bug, it was just set up like that. i.e. no thought to multi window/edit usage. 
Could there be some sort of 'keep alive' to resource locks from those edit windowswhile they are open. Then an auto clear of stale locks when not 'kept alive'? Maybe that would be too resource intensive?

**Comment by Deesen** [comment]
There are already 2 "keep alive"-functions.. one for the session and one for the mail-symbol for incoming mails. So it seems not to be that resource intensive. Maybe it is possible to modify the main session-function so it can keep track which pop-ups are open or already closed.

**Comment by Deesen** [comment]
@bossloper Just wanted to inform you that the actual solution of keeping resources locked is managed now as per your proposal. Thanks for it! I tried everything like using localStorage, or Javascript-solutions to collect events from different windows into mainframe, to send all locked IDs with a single request. But the only relyable cross-browser solution was the one you proposed at the very beginning ;-). Closing this one now.
